### ISSUE TITLE
Issue with restarting runs from checkpoint mid run (I believe)?

### ISSUE THREAD

**Issue by shadowbourne** [janosh/tensorboard-reducer#8]
PS C:\Users\shadow\Desktop\tb> tb-reducer -i 'whole game no double*' -o 'whole game no double' --lax-steps --lax-tags
Traceback (most recent call last):
  File "c:\users\shadow\appdata\local\programs\python\python38\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\shadow\appdata\local\programs\python\python38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\shadow\AppData\Local\Programs\Python\Python38\Scripts\tb-reducer.exe\__main__.py", line 7, in <module>
  File "c:\users\shadow\appdata\local\programs\python\python38\lib\site-packages\tensorboard_reducer\main.py", line 102, in main
    events_dict = load_tb_events(
  File "c:\users\shadow\appdata\local\programs\python\python38\lib\site-packages\tensorboard_reducer\io.py", line 104, in load_tb_events
    return {key: pd.concat(lst, join="inner", axis=1) for key, lst in out_dict.items()}
  File "c:\users\shadow\appdata\local\programs\python\python38\lib\site-packages\tensorboard_reducer\io.py", line 104, in <dictcomp>
    return {key: pd.concat(lst, join="inner", axis=1) for key, lst in out_dict.items()}
  File "c:\users\shadow\appdata\local\programs\python\python38\lib\site-packages\pandas\core\reshape\concat.py", line 298, in concat
    return op.get_result()
  File "c:\users\shadow\appdata\local\programs\python\python38\lib\site-packages\pandas\core\reshape\concat.py", line 516, in get_result
    indexers[ax] = obj_labels.get_indexer(new_labels)
  File "c:\users\shadow\appdata\local\programs\python\python38\lib\site-packages\pandas\core\indexes\base.py", line 3171, in get_indexer
    raise InvalidIndexError(
pandas.errors.InvalidIndexError: Reindexing only valid with uniquely valued Index objects

[files1.zip](https://github.com/janosh/tensorboard-reducer/files/6602534/files1.zip)

Help would be much appreciated, as I'm using this for my dissertation which is due in a couple of weeks! Thanks, and keep up the good work!

**Comment by janosh** [comment]
@shadowbourne Hey there. I don't think this one is an issue with TensorBoard Reducer. Your event files seem to have duplicate recordings for a few steps somewhere in the middle:

```py
from tensorboard_reducer import load_tb_events

out_dict = load_tb_events(
    "whole game no double*", strict_steps=False, strict_tags=False
)

(dupe_mask := (df := out_dict['eval-SuperMarioBros-1-1-v1'][0]).index.duplicated()).sum()
>>> 26
```

```py
df[dupe_mask].index
>>> 
df[dupe_mask].index
Int64Index([50550, 50600, 50650, 50700, 50750, 50800, 50850, 50900, 50950,
            51000, 51050, 51100, 51150, 51200, 51250, 51300, 51350, 51400,
            51450, 51500, 51550, 51600, 51650, 51700, 51750, 51800],
           dtype='int64', name='step')
```

We could add another kwarg `keep_dupes='first'|'last'` to `load_tb_events` to define which steps should be kept in case of duplicates. But then I also don't want to add too much complexity to that function. Strictly speaking, removing duplicate steps would be a preprocessing step to TensorBoard Reducer.

**Comment by janosh** [comment]
Actually, considering there's an [open GitHub as well as internal TensorBoard issue](https://github.com/tensorflow/tensorboard/issues/3570) about handling duplicate steps, maybe this is a common enough use case to warrant a new load option.

I added `handle_dup_steps=None` to `load_tb_events()`. [Released as `tensorboard-reducer==0.2.2`](https://pypi.org/project/tensorboard-reducer/0.2.2/). So could you once again (#7)  try `pip install -U tensorboard-reducer` followed by `tb-reducer -i 'seed?/*' -o out --lax-steps --handle-dup-steps keep-first` and see if that works? (Other possible values are `keep-last` and `mean`.)

**Comment by shadowbourne** [comment]
Hey that yeah I agree about it kind of being preprocessing, however I think you'll find a vast majority of the target user-base of this module are using it because they want to completely avoid having to deal with learning and using tensorboard, aka avoid any pre-processing. 
And I also think that, especially in deep and reinforcement learning, restarting training from checkpoints is an extremely common practice especially when using slurm jobs with limits, and storing checkpoints every "epoch" is extremely expensive, thus upon restart you end up with some overlap.
Therefore I think your decision to include this argument is an extremely good idea!
I also like your inclusion of the mean option, well done and thank you once again!
